### ISSUE TITLE
[BUG] Investigate unhandled promise in ProposerSetUpdate e2e tests

### ISSUE THREAD

**Issue by dutterbutter** [webb-tools/dkg-substrate#278]
**Describe the bug**

We need to investigate the ProposerSetUpdate e2e tests to find out where we are not handling a promise correctly. The ProposerSetUpdate e2e tests should be deterministic during ci runs and not timeout.

**To Reproduce**
1. Run `cargo build --release -p dkg-standalone-node --features integration-tests`
2. Run `cd dkg-test-suite && git submodule update --init --recursive`
3. Run `cd dkg-test-suite`
4. Run `yarn install`
5. To run all tests: `yarn test`

**Log output**

<details>
  <summary>Log Output</summary>
  
  ```Paste log output here
   1 failing

  1) E2E Test Runner
       Proposer Set Update Proposal
         proposer set update test:
     Error: Timeout of 300000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/home/runner/work/dkg-substrate/dkg-substrate/dkg-test-suite/tests/proposals.test.ts)
      at listOnTimeout (node:internal/timers:559:17)
      at processTimers (node:internal/timers:502:7)
  ```
</details> 

**Expected behaviour**

ProposerSetUpdate e2e tests should be deterministic and properly resolve promises to avoid timeouts. 

**Other information and links**

CI runner: https://github.com/webb-tools/dkg-substrate/runs/5755473503?check_suite_focus=true

**Comment by drewstone** [comment]
I think a potential reason this issue is occurring is because of the following:
- The keygen for the genesis DKG finishes.
- The authorities of the genesis set begin signing the first unsigned proposal (since `ProposerSetUpdate` is immediately created on new/genesis session)
- The public key of the DKG has not yet been added to the chain as we undertake a majority gossip protocol to push it on-chain.
- The signature of the `ProposerSetUpdate` hits the chain before the `DKGPublicKey` does and so we get an error.

**Comment by dutterbutter** [comment]
Consider moving ProposerSetUpdate at the same time as RefreshProposer is triggered.

**Comment by drewstone** [comment]
Or we can handle this as part of the worker, to ensure the worker waits to see the key on-chain. I will add this in my PR.

**Comment by dutterbutter** [comment]
Closed in #281
